### PR TITLE
40 Refactoring

### LINKED ISSUE #40: Refactoring Sprint3/4
- Es gibt hier noch das Problem, dass das erste Clipping aus storage entfernt wird, das gleich zu selectedClipping ist. Wenn wir also mehrmals das gleiche Item kopiert haben, wird unter Umständen nicht das richtige entfernt.  Ihr könnt aber auchn icht einfach "removeAt" nehmen, weil ja der Index hier von dem Index in storage unterschiedlich sein kann. Da müssen wir uns noch eine Lösung ausdenken.

- Das zweite Problem können wir unter Umständen lösen, indem wir in RCPluggableListMorph speichern, ob das letzte Event ein Maus oder Tastaturevent war und das dann ClippingsList mitteilen. Und dann führen wir diese Logik nur aus, wenn es ein Mausevent war. (gibt aber wahrscheinlich bessere Lösung)

- Window und Context Menu gemeinsame Oberklasse einführen

- Alles formattieren nach Coding standards

- Kaskade in "ClippingListTest>>testAppliesRegexFilter" statt "add...add...add"

- at: 1 zu first (in RCStorageTest>>testGetAllReturnsAllItems)

### PR DESCRIPTION
**Pull request by SiddeshkanthLogonathan**
Closes #40 

- [x] High Test-Coverage (90%), GUI Code might be an exception
- [x] All Tests passing
- [x] All files autoformatted (to be discussed)
- [x] Codequality: not many long methods
- [x] Codequality: Encapsulation (classes should be loosely coupled, law of demeter,...)
- [x] Timetracking: Issue Timetracking is complete (Time spent)

### CODE REVIEW

**Comment by SiddeshkanthLogonathan** [comment]
**We haven't completed:**
1. Formatting all methods
2. Merging RCWindow and RCContextMenu

File: RichClipboard.package/RCClippingsList.class/instance/lastEventWasMouseEvent..st [hunk 1]
```diff
@@ -0,0 +1,3 @@
+accessing
+lastEventWasMouseEvent: aBool
+	lastEventWasMouseEvent := aBool
```
Comment (jffjhnsn) [inline RichClipboard.package/RCClippingsList.class/instance/lastEventWasMouseEvent..st]:
```suggestion
lastEventWasMouseEvent: aBoolean
	lastEventWasMouseEvent := aBoolean
```

File: RichClipboard.package/RCClippingsList.class/instance/selectedClippingIndex..st [hunk 2]
```diff
@@ -2,7 +2,7 @@ accessing
 selectedClippingIndex: anIndex
 	"At this moment we do not want any item to be selected"
```
Comment (jffjhnsn) [inline RichClipboard.package/RCClippingsList.class/instance/selectedClippingIndex..st]:
```suggestion
	" At this moment we do not want any item to be selected "
	
```

File: RichClipboard.package/RCClippingsList.class/instance/selectedClippingIndex..st [hunk 3]
```diff
@@ -2,7 +2,7 @@ accessing
...
 	self changed: #selectedClippingIndex.
-	anIndex > 0 ifTrue: [
+	(anIndex > 0 and: [self lastEventWasMouseEvent]) ifTrue: [
```
Comment (jffjhnsn) [inline RichClipboard.package/RCClippingsList.class/instance/selectedClippingIndex..st]:
```suggestion
	(anIndex > 0 and: [ self lastEventWasMouseEvent ]) ifTrue: [
```

File: RichClipboard.package/RCPluggableListMorph.class/instance/keyStroke..st [hunk 4]
```diff
@@ -0,0 +1,4 @@
+event handling
+keyStroke: evt 
+	self listMorph lastEventWasMouseEvent: false.
+	super keyStroke: evt
```
Comment (jffjhnsn) [inline RichClipboard.package/RCPluggableListMorph.class/instance/keyStroke..st]:
```suggestion
keyStroke: anEvent 
	self listMorph lastEventWasMouseEvent: false.
	super keyStroke: anEvent
```

File: RichClipboard.package/RCPluggableListMorph.class/instance/mouseUp..st [hunk 5]
```diff
@@ -0,0 +1,4 @@
+event handling
+mouseUp: evt 
+	self listMorph lastEventWasMouseEvent: true.
+	super mouseUp: evt
```
Comment (jffjhnsn) [inline RichClipboard.package/RCPluggableListMorph.class/instance/mouseUp..st]:
```suggestion
mouseUp: anEvent
	self listMorph lastEventWasMouseEvent: true.
	super mouseUp: anEvent
```

**Review by jffjhnsn — CHANGES_REQUESTED** [review]
Styling changes

**Review by jffjhnsn — APPROVED** [review]

**Review by zortax — APPROVED** [review]

**Pull request merged**